packages/@okta/vuepress-site/docs/guides/mobile-idx-sdk-overview/main/android/gettingatoken.md

```diff
@@ -0,0 +1,24 @@
+```kotlin
+private suspend fun handleResponse(response: IdxResponse) {
+    // Check if sign-in is successful.
+    if (response.isLoginSuccessful) {
+        // Exchange the sign-in session token for a connection token.
+        when (val exchangeCodesResult = client?.exchangeInteractionCodeForTokens(response.remediations[ISSUE]!!)) {
+            is IdxClientResult.Error -> {
+                // Handle error.
+            }
+            is IdxClientResult.Success -> {
+                // Handle success.
+                // The token is in `exchangeCodesResult.result`.
+                // Store it securely for future use.
+            }
+            else -> {
+                // Handle error.
```

Comment:
If we make the `client` not nullable, this won't be possible, which would simplify this code block.

Reply:
Unfortunately this has to be written for what exists now. When the client is non-nullable we can revise the snippet.